app/components/branch-row.js

```diff
@@ -19,14 +20,20 @@ export default Component.extend({
   isTriggering: false,
   hasTriggered: false,
 
-  commitUrl: computed('branch.repository.{slug,vcsType}', 'branch.last_build.commit.sha', function () {
-    const [owner, repo] = this.get('branch.repository.slug').split('/');
-    const vcsType = this.get('branch.repository.vcs_type');
+  commitUrl: computed('branch.repository.id', 'branch.last_build.commit.sha', function () {
+    const repository = this.store.peekRecord('repo', this.get('branch.repository.id'));
+    const owner = repository.owner.login;
+    const repo = repository.name;
+    const vcsType = repository.vcsType;
     const commit = this.get('branch.last_build.commit.sha');
-
     return this.externalLinks.commitUrl(vcsType, { owner, repo, commit });
   }),
 
+  vcsType: computed('branch.repository.id', function () {
+    const repository = this.store.peekRecord('repo', this.get('branch.repository.id'));
+    return repository.vcsType;
```

Comment:
Same here, if the store content will change, the computed property won't be notified and won't recompute.